platform/web/os_web.cpp

```diff
@@ -157,7 +158,22 @@ void OS_Web::vibrate_handheld(int p_duration_ms) {
 }
 
 String OS_Web::get_user_data_dir() const {
-	return "/userfs";
+	String userfs = "/userfs";
+	String appname = get_safe_dir_name(GLOBAL_GET("application/config/name"));
+	if (!appname.is_empty()) {
+		bool use_custom_dir = GLOBAL_GET("application/config/use_custom_user_dir");
+		if (use_custom_dir) {
+			String custom_dir = get_safe_dir_name(GLOBAL_GET("application/config/custom_user_dir_name"), true);
+			if (custom_dir.is_empty()) {
+				custom_dir = appname;
+			}
+			return userfs.path_join(custom_dir).replace("\\", "/");
+		} else {
+			return userfs.path_join(get_godot_dir_name()).path_join("app_userdata").path_join(appname).replace("\\", "/");
+		}
+	}
+
+	return userfs.path_join(get_godot_dir_name()).path_join("app_userdata").path_join("[unnamed project]");
```

Comment:
Maybe we should move this logic in `OS::get_user_data_dir()` and remove it from Linux, Windows, and here? And just set `get_data_path()` to `/userfs` here. (The only difference between Linux and Windows implementations seems to be the `replace("\\", "/")` from Windows, which you're also using here. They probably don't hurt to have applied for Linux too?)

Unless `get_data_path()` should say `/home/web_user/.local/share` here and thus we can't rely on a common implementation using `get_data_path()`. In that case, maybe the whole appname/custom dir logic could be moved to a helper method in `ProjectSettings`?

Reply:
Yeah, I feel like at least the appname/custom dir logic should be moved, I'll look into it.